docs/reference/job-schema.rst

```diff
@@ -0,0 +1,84 @@
+Test job schema
+=================
+
+Test jobs can be defined in either YAML or JSON format.
+
+The following table lists the key elements that a job definition file should contain.
+
+.. list-table:: Test job schema
+  :header-rows: 1
+
+  * - Field
+    - Type 
+    - Description
+  * - ``job_queue``
+    - String
+    - Name of the job queue to which you want to submit the job. This field is mandatory.
+  * - ``global_timeout``
+    - integer
+    - | (Optional) Maximum time (in seconds) a job is allowed to run. If the global timeout is reached while a job is still running, Testflinger will close the job immediately. By default, the timeout for each Testflinger agent is 4 hours. 
+      | If your test job will presumably take more than 4 hours to complete, you can set a larger number in the ``global_timeout`` field in the test job. However, if a timeout is specified on a device, the value for this parameter will be overridden by the per-device configurations.
+  * - ``output_timeout``
+    - integer
+    - (Optional) Maximum time (in seconds) Testflinger should wait in the ``test`` phase to get output from the job. If the timeout is reached before the test phase has any output, Testflinger will cancel the job. This value should be smaller than, or equal to, the ``global_timeout``.  The default value is 15 minutes. 
+  * - ``allocation_timeout``
+    - integer
+    - (Optional) Maximum time (in seconds) Testflinger should wait in the ``allocate`` phase for multi-device jobs to reach the ``allocated`` state. If the timeout is reached before all devices are allocated, Testflinger will cancel the job. The default timeout to allocate all devices is 2 hours.
+  * - ``<phase>_data``
+    - dictionary
+    - | (Optional) Sections that define the configurations and commands about how the job should be executed in each of the test phases. If any phase encounters a non-zero exit code, no further phases will be executed.
+      | Supported test phases include: 
+      |   - provision
+      |   - firmware_update
+      |   - test
+      |   - allocate
+      |   - reserve 
+      | For detailed information about how to define the data to include in each test phase, see :doc:`test-phases`. 
+
+Environment variables
+----------------------------
+
+The following environment variables are supported in the test environment on the HOST system (not on the device under test). You can refer to these variables when defining a job:
```

Comment:
This is not quite right - and it's very specific to our labs. Even then, I don't think the server cert lab has this wifi data in their agents, just DEVICE_IP. There's nothing in testflinger that requires or restricts any values from the environment settings.

I think we could put something like this in the internal documentation, but  even then, we would have to denote which labs it even applies to.

Reply:
removed this section in appended commit